runtime/queries/metricsview_aggregation_test.go

```diff
@@ -2292,6 +2293,297 @@ measures:
+		},
+		ComparisonTimeRange: &runtimev1.TimeRange{
+			Start: timestamppb.New(time.Date(2022, 1, 3, 0, 0, 0, 0, time.UTC)),
+			End:   timestamppb.New(time.Date(2022, 1, 5, 0, 0, 0, 0, time.UTC)),
+		},
+		Limit:          &limit,
+		SecurityClaims: testClaims(),
+	}
+	err := q.Resolve(context.Background(), rt, instanceID, 0)
+	require.NoError(t, err)
+	require.NotEmpty(t, q.Result)
+	fields := q.Result.Schema.Fields
+	require.Equal(t, "pub,dom,timestamp,timestamp_year,measure_0,measure_1,m1,measure_0__p", columnNames(fields))
+	i := 0
+
+	for _, sf := range q.Result.Schema.Fields {
+		fmt.Printf("%v ", sf.Name)
+	}
+	fmt.Printf("\n")
+
+	for i, row := range q.Result.Data {
+		for _, sf := range q.Result.Schema.Fields {
+			fmt.Printf("%v ", row.Fields[sf.Name].AsInterface())
+		}
+		fmt.Printf(" %d \n", i)
+
+	}
+	rows := q.Result.Data
+	require.Equal(t, 2, len(rows))
+
+	i = 0
+	require.Equal(t, "Google,news.google.com,2022-01-01T00:00:00Z,2022-01-01T00:00:00Z,187.00,183.00,3.55,3.55", fieldsToString2digits(rows[i], "pub", "dom", "timestamp", "timestamp_year", "measure_0", "measure_0__p", "measure_1", "m1", "timestamp__previous", "timestamp_year__previous"))
+}
+
+func TestMetricsViewsAggregation_comparison_multiple_time_dim_cross_max_grain_boundaries(t *testing.T) {
+	rt, instanceID := testruntime.NewInstanceForProject(t, "ad_bids")
+
+	limit := int64(10)
+	q := &queries.MetricsViewAggregation{
+		MetricsViewName: "ad_bids_metrics",
+		Dimensions: []*runtimev1.MetricsViewAggregationDimension{
+			{
+				Name: "pub",
+			},
+			{
+				Name: "dom",
+			},
+
+			{
+				Name:      "timestamp",
+				TimeGrain: runtimev1.TimeGrain_TIME_GRAIN_HOUR,
+			},
+			{
+				Name:      "timestamp",
+				TimeGrain: runtimev1.TimeGrain_TIME_GRAIN_DAY,
+				Alias:     "timestamp_day",
+			},
+		},
+		Measures: []*runtimev1.MetricsViewAggregationMeasure{
+			{
+				Name: "measure_0",
+			},
+			{
+				Name: "measure_0__p",
+				Compute: &runtimev1.MetricsViewAggregationMeasure_ComparisonValue{
+					ComparisonValue: &runtimev1.MetricsViewAggregationMeasureComputeComparisonValue{
+						Measure: "measure_0",
+					},
+				},
+			},
+		},
+		Where: expressionpb.AndAll(
+			expressionpb.Eq("pub", "Google"),
+			expressionpb.Eq("dom", "news.google.com"),
+		),
+		Having: expressionpb.Gt("measure_0__p", 0.0),
+		Sort: []*runtimev1.MetricsViewAggregationSort{
+			{
+				Name: "pub",
+			},
+			{
+				Name: "dom",
+			},
+			{
+				Name: "timestamp",
+			},
+			{
+				Name: "timestamp_day",
+			},
+			{
+				Name: "measure_0",
+			},
+		},
+
+		TimeRange: &runtimev1.TimeRange{
+			Start: timestamppb.New(time.Date(2022, 1, 1, 23, 0, 0, 0, time.UTC)),
+			End:   timestamppb.New(time.Date(2022, 1, 2, 2, 0, 0, 0, time.UTC)),
+		},
+		ComparisonTimeRange: &runtimev1.TimeRange{
+			Start: timestamppb.New(time.Date(2022, 1, 2, 1, 0, 0, 0, time.UTC)),
+			End:   timestamppb.New(time.Date(2022, 1, 2, 3, 0, 0, 0, time.UTC)),
+		},
```

Comment:
I don't think this test case addresses the concern described in this comment: https://github.com/rilldata/rill/pull/5475#pullrequestreview-2266024728?

The goal is not to test overlapping base and comparison time ranges. The goal is to test that if the comparison time range has `start: "2023-11-01"` and the base time range has `start: "2024-02-01"`, then it should correct the comparison values for e.g. `2024-01-15` to `2024-04-15`, not `2025-04-15` (accidentally adding a year, which is the year-interval datediff between the two start times).

Reply:
We fixed the bug where we didn't subtract anything from larger grain times. I abstract language:
```
Assuming format yy-mm and base and comparison values:
base: [23-12 24-01 24-02]
 base_year: [23 24 24]

comparison_month: [24-02 24-03 24-05]
  comparison_year: [24 24 24]

date_diff(23-12,24-02,month) = 2
new_comparison = comparison - date_diff
new_comparison_year = date_trunc('year', comparison - date_diff))

Subtracting date diff from comparison:
new_comparison: [23-12 24-01 24-02]
  new_comparison_year: [23 24 24]
```
The same will work for hour and day and any other combinations. Not sure I'm following.

Reply:
Oh, I see what you mean, let me check.

Reply:
It's irrelevant, we don't check that record, it's a test typo - fixed.